staging/src/k8s.io/apiserver/pkg/server/options/encryption_provider_config.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	yaml "github.com/ghodss/yaml"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage/value"
+	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
+)
+
+// Used for parsing command line parameters for selecting transformer
+type EncryptionProviderConfig struct {
+	TransformerOverrides *map[schema.GroupResource]value.Transformer
+}
+
+func (e EncryptionProviderConfig) Set(filepath string) error {
+	f, err := os.Open(filepath)
+	if err != nil {
+		return fmt.Errorf("error opening encryption provider configuration file %q: %v", filepath, err)
+	}
+	defer f.Close()
+
+	if err := ConfigToTransformerOverrides(f, e.TransformerOverrides); err != nil {
+		return fmt.Errorf("error parsing encryption provider configuration file %q: %v", filepath, err)
+	}
+	return nil
+}
+
+func (e EncryptionProviderConfig) String() string {
+	return ""
+}
+
+func (e EncryptionProviderConfig) Type() string {
+	return "experimental-encryption-provider-config"
+}
+
+// ConfigToTransformerOverrides consumes an io.Reader containing a configuration file, and stores
+// the parsed encryption provider configuration to destination
+func ConfigToTransformerOverrides(f io.Reader, destination *map[schema.GroupResource]value.Transformer) error {
+	config, err := ioutil.ReadAll(f)
+	if err != nil {
+		return fmt.Errorf("could not read contents: %v", err)
+	}
+
+	var providers []map[string]interface{}
+	yaml.Unmarshal(config, &providers)
+
+	resourceToPrefixTransformer := map[schema.GroupResource][]value.PrefixTransformer{}
+
+	// For each provider listed in config file
+	for _, provider := range providers {
+		// Parse the config map to get configuration
+		providerConfig, err := parseProviderInfo(provider)
+		if err != nil {
+			return err
+		}
+		if providerConfig.Kind == "k8s-aes-gcm" {
```

Comment:
i'd say `k8s-aead-aes-gcm`, or we pick a colloquial name like `k8s-aes` and potentially version it in the future (for instance, if we add secret box, it might be `k8s-chacha`, and if we move to AES-GCM-SIV, it might be `k8s-aes-2`).  I think it's important to be specific about the choices, but most people don't understand the tradeoffs well enough that the additional info helps.